src/Cake/Infrastructure/Composition/ModuleLoader.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Autofac;
+using Cake.Core.Composition;
+
+namespace Cake.Client.Infrastructure.Composition
+{
+    public sealed class ModuleLoader
+    {
+        private readonly IContainer _container;
+
+        public ModuleLoader(IContainer container)
+        {
+            _container = container;
+        }
+
+        public IEnumerable<ICakeModule> LoadModules(IEnumerable<Type> types)
+        {
+            foreach (var type in types)
+            {
+                var module = LoadModule(type);
+                if (module != null)
+                {
+                    yield return module;
+                }
+            }
+        }
+
+        private ICakeModule LoadModule(Type type)
+        {
+            var constructor = GetGreediestConstructor(type);
+            var parameters = constructor.GetParameters();
+
+            var arguments = new object[parameters.Length];
+            for (int index = 0; index < parameters.Length; index++)
+            {
+                var parameter = parameters[index];
+                arguments[index] = _container.Resolve(parameter.ParameterType);
+            }
+
+            if (Activator.CreateInstance(type, arguments) is ICakeModule module)
+            {
+                return module;
+            }
+
+            return null;
+        }
+
+        private static ConstructorInfo GetGreediestConstructor(Type type)
```

Comment:
Replace with suggestion from @Meir017

Reply:
@Meir017 Remeber what your suggestion was here? I can't seem to find it 😢

Reply:
I think this:
```cs
private static ConstructorInfo GetGreediestConstructor(Type type) => type.GetTypeInfo()
    .GetConstructors()
    .OrderByDescending(ctor => ctor.GetParameters().Length)
    .First();
```